core/src/main/java/org/commonjava/maven/ext/core/impl/NexusStagingMavenPluginRemovalManipulator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2012 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.maven.ext.core.impl;
+
+import org.commonjava.maven.ext.common.model.Project;
+import org.commonjava.maven.ext.core.ManipulationSession;
+import org.commonjava.maven.ext.core.state.NexusStagingMavenPluginRemovalState;
+import org.commonjava.maven.ext.core.state.PluginState;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * {@link Manipulator} implementation that can remove nexus-staging-maven-plugin from a project's pom file.
+ * Configuration is stored in a {@link PluginState} instance, which is in turn stored in the
+ * {@link ManipulationSession}.
+ */
+@Named("nexus-staging-maven-plugin-removal-manipulator")
+@Singleton
+public class NexusStagingMavenPluginRemovalManipulator
+        extends PluginRemovalManipulator {
+    private ManipulationSession session;
```

Comment:
Remove 'this' session and use the inherited one

Reply:
Done.